packages/compass-generative-ai/src/atlas-ai-service.ts

```diff
@@ -204,9 +210,65 @@ const aiURLConfig = {
   cloud: {
     aggregation: (groupId: string) => `ai/v1/groups/${groupId}/mql-aggregation`,
     query: (groupId: string) => `ai/v1/groups/${groupId}/mql-query`,
+    'mock-data-schema': (groupId: string) =>
+      `ai/v1/groups/${groupId}/mock-data-schema`,
   },
 } as const;
-type AIEndpoint = 'query' | 'aggregation';
+
+export const MockDataSchemaRawFieldMappingShape = z.record(
+  z.string(),
+  z.object({
+    type: z.string(),
+    sampleValues: z.array(z.unknown()).optional(),
+    probability: z.number().min(0).max(1).optional(),
+  })
+);
+
+export const MockDataSchemaRequestShape = z.object({
+  collectionName: z.string(),
+  databaseName: z.string(),
+  schema: MockDataSchemaRawFieldMappingShape,
+  validationRules: z.record(z.string(), z.unknown()).nullable().optional(),
+  includeSampleValues: z.boolean().default(false),
+});
```

Comment:
Why is this defined as a zod schema if it's not being used?

Reply:
Also not sure why we would want zod validation / parsing for input that we are in full control of 🤔